src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep.java

```diff
@@ -92,6 +94,8 @@
                 MultiBinding.MultiEnvironment environment = binding.bind(run, workspace, launcher, listener);
                 unbinders.add(environment.getUnbinder());
                 overrides.putAll(environment.getValues());
+                IdCredentials id = CredentialsProvider.findCredentialById(binding.getCredentialsId(), binding.type(), run);
+                CredentialsProvider.track(run, id);
```

Comment:
So track(Node, C) is to track when the credential is used *to launch that node*. Not the case here. We already have tracking between the Run and the Node(s) it ran on.